### ISSUE TITLE
broken types for release groups

### ISSUE THREAD

**Issue by selfisekai** [Borewit/musicbrainz-api#627]
query:
```js
mb.browseReleaseGroups({ artist: '1c25cfa7-037e-4d82-a7b9-1df62fb6d9c7' })
```

returns this object:
```ts
export interface IBrowseReleaseGroupsResult {
    'release-groups': IReleaseGroupsQuery[];
    'release-group-count': number;
    'release-group-offset': number;
}
```

the `'release-groups'` property is declared to be of this type:
```ts
export interface IReleaseGroupsQuery extends IPagination {
    artist?: string;
    collection?: string;
    release?: string;
}
```

however, the actual object looks like this - nothing from the typedef matches:
```js
    {
      disambiguation: '',
      id: '00080a9c-11ff-41b0-b325-454bcc16be83',
      'secondary-type-ids': [],
      'secondary-types': [],
      title: 'Śpiewnik domowy',
      'primary-type': 'EP',
      'first-release-date': '2020-04-17',
      'primary-type-id': '6d0c5bf6-7a33-3420-a519-44fc63eedebf'
    },
```

**Comment by Haschikeks** [comment]
Hi @selfisekai,

thanks for reporting this.
Unfortunately the wrong type is used here. It probably should be `IReleaseGroup`, but there also seems to be some inconsistencies about the fields.

This is the current type definition, which does not fully match the response from the server.
```typescript
export interface IReleaseGroup extends IEntity {
  count: number;
  title: string;
  'primary-type': string;
  'sort-name': string;
  'artist-credit': { artist: IArtist, name: string, joinphrase: string }[];
  releases?: IRelease[]; // include 'releases'
}
```

The same problem seem to be present on the `IBrowseSeriesResult` & `IBrowseWorksResult` types as well.

You are welcome to submit a PR to fix the typings. I might only be able to attend this later next week.

**Comment by Borewit** [comment]
Resolved in [v0.10.1](https://github.com/Borewit/musicbrainz-api/releases/tag/v0.10.1)
